### PR TITLE
SAP Systems list revamp

### PR DESCRIPTION
**Pull request by fabriziosestito**
SAP Systems list revamp, now we only list the application sap systems and the related application instances
Still missing: Health container & filters (under discussion), tags

A new viewmodel sap systems table was created to simplify the KV store data processing.

### CODE REVIEW

**Comment by fabriziosestito** [comment]
image

File: web/environments.go [hunk 1]
```diff
@@ -102,29 +106,140 @@ func NewLandscapeHandler(client consul.Client) gin.HandlerFunc {
...
+	}
+
+	for _, row := range rowsBySID {
+		sapSystemsTable = append(sapSystemsTable, row)
+	}
+
+	return sapSystemsTable
+}
+
+func (t SAPSystemsTable) filter(sid []string) SAPSystemsTable {
```
Comment (arbulu89) [inline web/environments.go:184]: Maybe `filterBySid` would be a better name

File: web/templates/sapsystems.html.tmpl [hunk 2]
```diff
@@ -1,10 +1,94 @@
...
+                                <th scope='col'>Cluster name</th>
+                                <th scope='col'>Host</th>
+                            </tr>
+                            </thead>
+                            <tbody>
+                            {{- range .InstancesTable }}
+                            <tr>
+                                <td class="row-status">{{ template "health_icon" "unknown" }}</td>
+                                <td>
+                                    <a href="/sapsystems/{{ .SID }}?environment=default&landscape=default">{{ .SID }}</a>
```
Comment (arbulu89) [inline web/templates/sapsystems.html.tmpl:76]: We will need to be careful with this `SID` link (in the inner table)
This SID could be HANA installation SID, which is not a SAP system. 
We would need to see what to do. 
Maybe, we should remove the link from the internal table

**Review by arbulu89 — APPROVED** [review]
Good work!

File: web/environments.go (hunk 1 above)
Comment (fabriziosestito) [reply web/environments.go:184]: we will need to filter by health and possibly tag in the near future, i would leave it as `filter` also to match the `ClustersTable` filter function

File: web/templates/sapsystems.html.tmpl (hunk 2 above)
Comment (fabriziosestito) [reply web/templates/sapsystems.html.tmpl:76]: if we remove the link and the HANA SID is different from the outer row, we would not have a link to the HANA SID

Comment (arbulu89) [reply web/templates/sapsystems.html.tmpl:76]: Yes, I know. What this page should redirect to SAP systems, and a HANA installation is not a full SAP system.
I don't know if we will have pages for individual SAP instances or databases honestly

Comment (fabriziosestito) [reply web/templates/sapsystems.html.tmpl:76]: Ok I was missing this context, i will remove the link in the inner rows then

**Review by arbulu89 — APPROVED** [review]

**Pull request merged**